test/remote/gateways/remote_elavon_test.rb

```diff
@@ -424,6 +424,21 @@ def test_successful_purchase_with_truncated_data
     assert_success response
   end
 
+  def test_successful_special_character_encoding_truncation
+    special_card = credit_card('4000000000000002')
+    special_card.first_name = 'Fear & Loathing'
+    special_card.last_name = 'Castañeda'
+
+    response = @gateway.purchase(@amount, special_card, @options)
+
+    assert_success response
+    assert response.test?
+    assert_equal 'APPROVAL', response.message
+    assert_equal 'Fear & Loathin', response.params['first_name']
+    assert_equal 'Castañeda', response.params['last_name']
+    assert response.authorization
+  end
```

Comment:
Just to confirm - are we getting the gateway error for this test with the current adapter behavior on master?

Reply:
No, this is just testing if the gateway request was encoded/decoded correctly. I think that specific error resulted from an issue with invalid UTF-8 bytes being submitted thru iFrame and then .delete being called on an invalid byte. I will write a test for that!

Reply:
Possibly that error was also related to the iso-8859-1 forced encoding introducing an invalid UTF-8 byte